### PR TITLE
Simplify parsing of `AMENT_PREFIX_PATH`

### PR DESCRIPTION
**Pull request by haixuanTao**
This PR simplify how ROS2 messages are parsed using dora.

It will parse all messages included in the `AMENT_PREFIX_PATH` folder that has a path in the following format: `<package>/msg/<message>.msg`.

Similarly for services it will be parsed as: `<package>/srv/<service>.srv`.

It does not requires `.idl` anymore nor does it needs to be in the `ament_index`.

Fixes: #481

### CODE REVIEW

File: libraries/extensions/ros2-bridge/msg-gen/src/parser/package.rs [hunk 1]
```diff
@@ -1,94 +1,106 @@
...
+                        warn!(
+                        "found two versions of package: {:?}, message: {:?}. will skip the one in: {:#?}",
+                        package, file_name, path
+                    );
+                        continue;
+                    } else {
+                        visited_files.push((package.clone(), file_name.clone()));
+                    }
+
+                    match map.get_mut(&package) {
```
Comment (phil-opp) [inline libraries/extensions/ros2-bridge/msg-gen/src/parser/package.rs:63]: We could use the [entry API](https://doc.rust-lang.org/stable/std/collections/struct.HashMap.html#method.entry) here to avoid the duplication:

```rust
let p = map.entry(package.clone).or_insert_with(|| Package::new(package.clone()));
```

File: libraries/extensions/ros2-bridge/msg-gen/src/types/primitives.rs [hunk 2]
```diff
@@ -241,7 +241,7 @@ impl GenericString {
         // TODO: Assertion
         let value = Literal::string(value);
         if self.is_wide() {
-            quote! { ::widestring::U16String::from_str(#value) }
+            quote! { ffi::U16String::from_str(#value) }
```
Comment (phil-opp) [inline libraries/extensions/ros2-bridge/msg-gen/src/types/primitives.rs:244]: Could you clarify why this widestring change is required?

File: apis/python/node/test/test_dora.py
Comment (phil-opp) [inline apis/python/node/test/test_dora.py:1]: unrelated file

File: apis/python/node/generate_stubs.py
Comment (phil-opp) [inline apis/python/node/generate_stubs.py:1]: unrelated file

File: apis/python/node/dora/dora.pyi
Comment (phil-opp) [inline apis/python/node/dora/dora.pyi:1]: unrelated file

File: .github/install.sh
Comment (phil-opp) [inline .github/install.sh:1]: unrelated file

File: libraries/extensions/ros2-bridge/msg-gen/src/parser/package.rs (hunk 1 above)
Comment (haixuanTao) [reply libraries/extensions/ros2-bridge/msg-gen/src/parser/package.rs:63]: Indeed. Going to make the change.

**Review by phil-opp — APPROVED** [review]
Thanks!

File: libraries/extensions/ros2-bridge/msg-gen/src/types/primitives.rs (hunk 2 above)
Comment (haixuanTao) [reply libraries/extensions/ros2-bridge/msg-gen/src/types/primitives.rs:244]: So it seems that with the current `AMENT_PREFIX_PATH` we're parsing more messages in the CI/CD than before. Some of them containing U16String. 

It seemed that the previous implementation were dependent on a miss imported widestring package that cannot be found. 

I'm not exactly sure if there is another implementation to tokenize a U16 string into `chars: Vec>u16>` but this implementation keep the spirit of the previous implementation. But, ultimately, it seems that U16String is a bit experimental, when I search for the procedural macro calls.

**Comment by haixuanTao** [comment]
Thanks for catching the unwanted commit :pray:
